core/domain/exp_domain.py

```diff
@@ -4224,6 +4255,12 @@ def validate(self):
             raise Exception(
                 'Assigned state ids should be smaller than last state id used.')
 
+        if not all(isinstance(x, int) for x in state_ids):
```

Comment:
Can you add a test for this new validation check?

Reply:
Done